packages/react/src/FilteredActionList/FilteredActionListWithDeprecatedActionList.tsx

```diff
@@ -46,6 +48,68 @@ const StyledHeader = styled.div`
   z-index: 1;
 `
 
+const getItemWithActiveDescendant = (
+  listRef: React.RefObject<HTMLElement>,
+  items: FilteredActionListProps['items'],
+) => {
+  const listElement = listRef.current
+  const activeItemElement = listElement?.querySelector('[data-is-active-descendant]')
+
+  if (!listElement || !activeItemElement?.textContent) return
+
+  const optionElements = listElement.querySelectorAll('[role="option"]')
+
+  const index = Array.from(optionElements).indexOf(activeItemElement)
+  const activeItem = items[index]
+  const text = activeItem.text
+  const selected = activeItemElement.getAttribute('aria-selected') === 'true'
+
+  return {index, text, selected}
+}
+
+async function announceText(text: string) {
+  const liveRegion = document.querySelector('live-region')
+
+  liveRegion?.clear() // clear previous announcements
+
+  await announce(text, {
+    delayMs: 500,
+    from: liveRegion ? liveRegion : undefined, // announce will create a liveRegion if it doesn't find one
+  })
+}
+
+const announceItemsChanged = debounce(
+  async (items: FilteredActionListProps['items'], listContainerRef: React.RefObject<HTMLElement>) => {
+    const liveRegion = document.querySelector('live-region')
+
+    liveRegion?.clear() // clear previous announcements
+
+    // give @primer/behaviors a moment to update active-descendant
+    await new Promise(resolve => window.requestAnimationFrame(resolve))
+
+    const activeItem = getItemWithActiveDescendant(listContainerRef, items)
+    if (!activeItem) return
+    const {index, text, selected} = activeItem
+
+    const announcementText = [
+      'List updated',
+      `Focused item: ${text}`,
+      `${selected ? 'selected' : 'not selected'}`,
+      `${index + 1} of ${items.length}`,
+    ].join(', ')
+
+    await announce(announcementText, {
+      delayMs: 500,
+      from: liveRegion ? liveRegion : undefined, // announce will create a liveRegion if it doesn't find one
+    })
+  },
+  250,
+)
+
+async function announceFilterFocused() {
+  await announceText('Focus on filter text box and list of items')
+}
+
```

Comment:
moved all this logic that was making the SelectPanel component bulky down to the FilteredActionList so we can get rid of it when we finally remove it 💃